src/ucs/stats/stats.c

```diff
@@ -460,6 +474,171 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
+    uint32_t counter_index;
+    ucs_stats_aggrgt_sum_clsid_to_name_db_t *element;
+    ucs_stats_aggrgt_sum_counter_name_t     *element_cnt_names;
+    ucs_status_t status;
+
+    /* Get current array length */
+    db_size = ucs_array_length(&ucs_stats_context.aggrgt_sum_db);
+
+    status = ucs_array_append(aggrgt_sum_db_array_t, &ucs_stats_context.aggrgt_sum_db);
+    if (status != UCS_OK) {
+        ucs_assert(0);
+        return status;
+    }
+
+    element = ucs_array_last(&ucs_stats_context.aggrgt_sum_db);
+
+    /* Initialize entry */
+    offset = element->aggregate_sum_class_offset = 
+        ucs_array_length(&ucs_stats_context.aggrgt_cnt_names_db);
+
+    /* Update class_id in node structure */
+    node->cls->class_id = db_size;
+
+    ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+        ucs_list_for_each(temp_node, &filter_node->type_list_head,
+                          type_list) {
+            status = ucs_array_append(aggrgt_sum_db_array_counter_name_t,
+                                      &ucs_stats_context.aggrgt_cnt_names_db);
+            if (status != UCS_OK) {
+                ucs_assert(0);
+               return status;
+            }
+
+            element_cnt_names = ucs_array_last(&ucs_stats_context.aggrgt_cnt_names_db);
+
+            element_cnt_names->counter_name = node->cls->counter_names[counter_index];
+            element_cnt_names->class_name = node->cls->name;
+        }
+    }
+
+    return offset;
+}
+
+static uint32_t
+ucs_stats_clsid_to_base_offset(ucs_stats_node_t *node, size_t num_counters_in_class)
+{
+    size_t offset;
+    ucs_stats_aggrgt_sum_clsid_to_name_db_t *element;
+
+    pthread_mutex_lock(&ucs_stats_context.lock);
+
+    if (ucs_likely(node->cls->class_id != 0)) {
+        element = &ucs_array_elem(&ucs_stats_context.aggrgt_sum_db,
+                                  node->cls->class_id);
+        offset  = element->aggregate_sum_class_offset;
+    } else {
+        offset = ucs_stats_agrgt_sum_clsid_alloc(node, num_counters_in_class);
+    }
+
+    pthread_mutex_unlock(&ucs_stats_context.lock);
+
+    return offset;
+}
+
+
+/**
+ * Returns the aggregate_sum of all counters from all interfaces for a compact statistics trace.
+ * Used for Score-P UCX plugin to collect a compact trace of the counters.
+ *
+ * @param [in]  node                      Pointer to the statistics node.
+ * @param [in]  sel                       Whether or not to recursively iterate through the children.
+ * @param [in]  current_num_counters      Current number of counters iterated through (call with 0).
+ * @param [out] ucs_stats_aggregate_sum   The aggregate-sum of all counters of each class/type.
+ * @param [in]  num_stats_counters_max The number of elements in ucs_stats_aggregate_sum.
+ *
+ * Complexity: O(n)
+ *
+ * Returns: The total number of counters.
+ *
+ * Remarks: The function is recursive.
+ */
+static size_t ucs_stats_aggregate_sum_recurs(ucs_stats_node_t *node,
```

Comment:
this function should return void because its return value is not used

Reply:
Fixed.